src/control/lib/control/pool.go

```diff
@@ -681,3 +681,63 @@ func PoolReintegrate(ctx context.Context, rpcClient UnaryInvoker, req *PoolReint
 
 	return nil
 }
+
+type (
+	// PoolHandleCleanupReq contains the parameters for a pool cleanup request.
+	PoolHandleCleanupReq struct {
+		msRequest
+		unaryRequest
+		Handles map[string]map[string]bool //This format may seem awkward but it makes it possible to pass it directly from the process monitor
```

Comment:
The fact that you have to comment it makes me think perhaps it's the wrong type to use for this public API input. It should probably look more like your protobuf HandleList.

Reply:
Completely rewrote this so no longer necessary.